lib/archethic/networking/ip_lookup/nat_discovery/miniupnp.ex

```diff
@@ -43,8 +43,31 @@ defmodule Archethic.Networking.IPLookup.NATDiscovery.MiniUPNP do
     end
   end
 
+  @spec do_open_port(:inet.ip_address(), non_neg_integer()) :: :ok | :error
   defp do_open_port(local_ip, port) do
-    opts = [
+    with {reason, status} when status != 0 <- System.cmd(@upnpc, map_query(local_ip, port)),
```

Comment:
I think using the happy path firstly, we would make more sense.
The error branching should be done on the else, I think.

Reply:
We could use something like this:
```elixir
case System.cmd(@upnpc, map_query(local_ip, port)) do
  {_, 0} ->
    :ok 
  {reason, status} ->
    reason
    |> parse_reason()
    |> handle_error()
 end
```